.gitlab/build_quartz.yml

```diff
@@ -0,0 +1,52 @@
+##############################################################################
+# Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+# other Tribol Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (MIT)
+##############################################################################
+
+####
+# This is the shared configuration of jobs for quartz
+.on_quartz:
+  variables:
+    ALLOC_OPTIONS: "-p pdebug --exclusive --mpibind=off"
+    ALLOC_COMMAND: "salloc ${ALLOC_OPTIONS} -N ${ALLOC_NODES} -t ${ALLOC_TIME}"
+  tags:
+    - shell
+    - quartz
+  rules:
+    - if: '$CI_COMMIT_BRANCH =~ /_qnone/ || $ON_QUARTZ == "OFF"' #run except if ...
+      when: never
+    - when: on_success
```

Comment:
When a job has a `rules` section, the parsing will stop at the first conditional that resolves to "true" and apply the associated `when` clause.
Illustration:
- The above rules section means:
  1. If the branch name contains "_qnone" or the CI variable "ON_QUARTZ" is set to "OFF", then the job will never run. (That a way to run your pipeline with no access to quartz, for testing, for days when quartz is offline, or for a personal clone by an intern with partial access rights).
  2. else, run if previous jobs passed.
  
- No conditional resolves to "true" (see that has an "else"). If the parsing reaches a rule with no conditional, the subsequent rules will never be parsed. The following code means the job will run if the status of the required preceding jobs is success.
```
job:
  rules:
    - /* else */ when: on_success
    # it’s useless to add other rules here.
```
- When no rule section is provided, it’s equivalent to that last example: having a rule with `when: on_success`. Gitlab provides a shortcut to change this default: you can have the when clause directly in the job:
```
job:
  when: on_failure
```
- In a `rules` section, If no "true" statement is found, the job will not run. It’s like if there was a last rule: "-when: never".
- In Serac, rules are split in two job templates. GitLab will stack them depending on the order of definition (the order in which the templates where defined in the job, see the `extends` section). This is not recommended as this is brittle.
Instead, I recommend setting all the rules for all the jobs at once. An example in the radiuss-shared-ci implementation:
https://github.com/LLNL/radiuss-shared-ci/blob/c9c37efcd7eecfbdc99388ad8063daf4097bf788/ruby-build-and-test.yml#L16